pkg/releasing/git_changelog_private.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2021 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Utilities to extract git commit descriptions in useful ways."""
+
+import argparse
+import os
+import subprocess
+import sys
+
+
+def guess_previous_release_tag(git_path, pattern=None):
+  most_recent = None
+  cmd = [git_path, 'tag']
+  if pattern:
+    cmd.extend(['--list', pattern])
+  with subprocess.Popen(cmd, stdout=subprocess.PIPE) as proc:
+    most_recent = proc.stdout.read().decode('utf-8')
+    most_recent = most_recent.strip().replace('\n\n', '\n').split('\n')[-1]
```

Comment:
I did see multiple spaces in some cases, which is why I made this overly complex.
I'm trying to figure out what should happen on the edge cases. Or, wait until people start complaining about handling bad patterns to see the seriousness of the issue. I am inclined to do the latter, because I don't have a good enough sample of the kinds of tag names people create.

Eventually, I think this is going to have to be a full semver compare, but I don't want to write that in this CL because that is lower priority for me - and I can get community contributions for that.

Reply:
All fair, thanks for the information.  It may be worth a comment describing what is going on here, but otherwise, this is fine.

Reply:
ack